exchange-v2/contracts/ExchangeV2Core.sol

```diff
@@ -33,6 +34,65 @@ abstract contract ExchangeV2Core is Initializable, OwnableUpgradeable, AssetMatc
         emit Cancel(orderKeyHash);
     }
 
+    /**
+     * @dev function, generate sellOrder and buyOrder from parameters and call validateAndMatch() for purchase transaction
+     * @param direct struct with parameters for buy operation
+     * @param sellData data type V3_SELLV for left order
+     * @param buyData data type 3_BUY for right order
+     */
+    function directPurchase(
+        LibDirectPurchase.Purchase memory direct,
+        bytes memory sellData,
+        bytes memory buyData
```

Comment:
think 1 data is enough, second one can be mirrored

Reply:
I think it`s true, but I have another task for it